### PR TITLE
PageStorage: Fix unstable rename test

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/5761

Problem Summary: as the issue descript

### What is changed and how it works?

* Disallow non-atomic rename table by `RUNTIME_ASSERT`
* Remove tests that perform non-atomic rename, rewrite them to test the logic of atomic rename
  * Rewrite `raft/schema/rename_tables.test` by `SchemaSyncTest.RenameTables`
  * Rewrite `raft/schema/partition_table_restart.test` by `SchemaSyncTest.PartitionTableRestart`
  * Rewrite a part of `raft/txn_mock/partition_table.test` by `SchemaSyncTest.RenamePartitionTable`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TiDB/Schema/tests/gtest_schema_sync.cpp [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        auto & flash_ctx = global_ctx.getTMTContext();
+        flash_ctx.getSchemaSyncer()->reset();
+    }
+
+    ManageableStoragePtr mustGetSyncedTable(TableID table_id)
+    {
+        auto & flash_ctx = global_ctx.getTMTContext();
+        auto & flash_storages = flash_ctx.getStorages();
+        auto tbl = flash_storages.get(table_id);
+        RUNTIME_CHECK(tbl, "Can not find table in TiFlash instance!", table_id);
```
Comment (breezewish) [inline dbms/src/TiDB/Schema/tests/gtest_schema_sync.cpp:125]:
```suggestion
        RUNTIME_CHECK_MSG(tbl, "Can not find table {} in TiFlash instance!", table_id);
```

Comment (JaySon-Huang) [reply dbms/src/TiDB/Schema/tests/gtest_schema_sync.cpp:125]: I get it. The error message is duplicated if we use RUNTIME_CHECK
`DB::Exception: Check tbl failed, "Can not find table in TiFlash instance!" = Can not find table in TiFlash instance!, table_id = 36`

**Review by flowbehappy — APPROVED** [review]

**Review by lidezhu — APPROVED** [review]
LGTM

**Pull request merged**